adascape/fastscape_ext.py

```diff
@@ -173,9 +174,20 @@ def run_step(self):
         self.abundance = self._model.abundance
         self._model.evaluate_fitness()
 
+    def _island_boundary(self):
+        xx, yy = np.meshgrid(self.grid_x, self.grid_y)
+        island_mask = self.topo_elevation > self.disp_boundary
+        island_grid = np.column_stack((xx[island_mask], yy[island_mask]))
+        cvhull = ConvexHull(island_grid)
+        return island_grid[cvhull.vertices]
+
     @xs.runtime(args='step_delta')
     def finalize_step(self, dt):
-        self._model.update_individuals(dt, self.disp_boundary)
+        if self.disp_boundary is not None:
+            boundary_points = self._island_boundary()
+            self._model.update_individuals(dt, boundary_points)
+        else:
+            self._model.update_individuals(dt, self.disp_boundary)
```

Comment:
```suggestion
            self._model.update_individuals(dt)
```

Makes it clearer that no boundary is applied.